docs/codeql/ql-language-reference/modules.rst

```diff
@@ -273,4 +273,69 @@ The ``<module_expression>`` itself can be a module name, a selection, or a quali
 reference. For more information, see ":ref:`name-resolution`."
 
 For information about how import statements are looked up, see "`Module resolution <https://codeql.github.com/docs/ql-language-reference/ql-language-specification/#module-resolution>`__"
-in the QL language specification. 
\ No newline at end of file
+in the QL language specification. 
+
+Built-in modules
+****************
+
+QL defines a `QlBuiltins` module, which is a built-in module that is always in scope.
+Currently, it defines a single parameterized sub-module
+`EquivalenceRelation`, that provides an efficient abstraction for working with
+(partial) equivalence relations in QL.
+
+Equivalence relations
+=====================
+
+The built-in `EquivalenceRelation` module is parameterized by a type `T` and a
+binary base relation `eq` on `T`. The symmetric and transitive closure of `eq`
+induces a partial equivalence relation on `T`. If `eq` is reflexive, then the
+induced relation is an equivalence relation on `T`.
+
+The `EquivalenceRelation` module exports a `getEquivalenceClass` predicate that
+gets the equivalence class, if any, associated with a given `T` element by the
+(partial) equivalence relation induced by `eq`.
+
+The following example illustrates an application of the `EquivalenceRelation`
+module to generate a custom equivalence relation on an :ref:`algebraic datatype <algebraic-datatypes>`:
+
+.. code-block:: ql
+
+    newtype Node = MkNode(int x) { x in [1 .. 6] }
```

Comment:
`newtype` is a somewhat under-documented feature, and users might want to learn about `EquivalenceRelation` without first having to learn about `newtype`. Can you change this into

```
class Node extends int {
  Node() { this in [1 .. 6] }
}
```
I forget whether a `toString` override is mandatory here. Hopefully not if you have `int`s in the `from` clause.

Reply:
Good call. Yes, it works without the `toString` because we are selecting `int`s.